packages/compiler/server/serverlib.ts

```diff
@@ -613,25 +613,26 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
-  async function getDocument(type: Type): Promise<string> {
+  async function getTypeDetails(type: Type): Promise<string> {
     return type.kind;
   }
 
-  async function getTypeDetails(params: HoverParams): Promise<Hover> {
-    const sym = await compile(params.textDocument, (program, document, file) => {
+  async function getHover(params: HoverParams): Promise<Hover> {
+    const typesym = await compile(params.textDocument, (program, document, file) => {
```

Comment:
the name of this variable here doesn't make much sense. Either you should have it return the type and call `getTypeDetails` after or this should just be `typeDetails`